modules/imap/site.js

```diff
@@ -881,8 +881,10 @@ var imap_setup_message_view_page = function(uid, details, list_path, callback) {
                         .append(msgResponse.msg_text)
                         .append(msgResponse.msg_parts);
         document.title = $('.header_subject th').text();
-        $('.header_subject th').append('<i class="bi bi-x-lg close_inline_msg"></i>');
-        $('.close_inline_msg').on("click", function() { msg_inline_close(); });
+        if ($('.header_subject th').find('.bi.bi-star-half.account_icon').length === 0) {
+            $('.header_subject th').append('<i class="bi bi-x-lg close_inline_msg"></i>');
+            $('.close_inline_msg').on("click", function() { window.history.back(); });
```

Comment:
there is a function `msg_inline_close` that can be used to close the inline, if you decide to not use you should then remove. But for me i would prefer to use the function instead of removing.